### PR TITLE
Internalize DDSketch implementation

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
There are attributes and methods on DDSketch that are needlessly public.
Internalizing them makes maintenance easier.

Depends on #38

### CODE REVIEW

File: releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+prelude: >
+  The public API
```
Comment (brettlangdon) [inline releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml:3]: how does this get rendered?

I don't think this gets associated with this upgrade note it just gets added at the top?

Comment (Kyle-Verhoog) [reply releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml:3]: Oh wait, that totally shouldn't be there lol.

Comment (Kyle-Verhoog) [reply releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml]: Should be fixed now

File: releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
+  - "``DDSketch`` attributes ``mapping``, ``store``, ``negative_store``, ``zero_count``, ``relative_accuracy``, ``min`` and ``max`` have been made internal."
+  - "``DDSketch.copy`` has been made internal."
+  - "``DDSketch.mergeable`` has been made internal."
```
Comment (brettlangdon) [inline releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml:5]: why do this as a string and not a multiline string?

File: releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
+  - "``DDSketch`` attributes ``mapping``, ``store``, ``negative_store``, ``zero_count``, ``relative_accuracy``, ``min`` and ``max`` have been made internal."
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml:3]: @brettlangdon I'm conflicted as to whether we should say removed vs "made internal". Any thoughts?

Comment (brettlangdon) [reply releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml:3]: I think "removed" and "made readonly" reads good?

File: releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml (hunk 2 above)
Comment (Kyle-Verhoog) [reply releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml:5]: Multiline doesn't seem to pick up on the bullet points

File: releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml (hunk 3 above)
Comment (Kyle-Verhoog) [reply releasenotes/notes/ddsketch-api-a84ffc0875bbacd6.yaml:3]: yeah I think I prefer "removed"

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**